gnosis/gp-v2-services
# Manage stdout/stderr split from Rust binary

Sends alerts to stderr from the Rust binary if log level is above a user-specified threshold.

It replaces `regex-stream-split` and makes the ad hoc Docker entry point redundant, which is why it's deleted in this PR. It should hopefully provide more reliable logging and should fix some intermittent alerts due to our use of `kill`.

Closes #1228.

### Test Plan

Manual tests: 
```
$ cargo --quiet run --bin solver -- --log-stderr-threshold info >/dev/null
2021-10-10T16:23:17.886Z  INFO solver: running solver with validated Arguments {
    shared: Arguments {
<snip>
$ cargo --quiet run --bin solver -- --log-stderr-threshold error >/dev/null
2021-10-10T16:22:29.620Z ERROR shared::tracing: thread 'main' panicked at 'No amount to estimate prices with set.', solver/src/main.rs:302:10:
thread 'main' panicked at 'No amount to estimate prices with set.', solver/src/main.rs:302:10
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
$ cargo --quiet run --bin solver -- --log-stderr-threshold off >/dev/null
thread 'main' panicked at 'No amount to estimate prices with set.', solver/src/main.rs:302:10
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
$ cargo --quiet run --bin solver -- --log-stderr-threshold off 2>/dev/null
2021-10-10T16:25:46.632Z  INFO solver: running solver with validated Arguments {
    shared: Arguments {
<snip>
$ cargo --quiet run --bin solver -- --log-stderr-threshold invalid >/dev/null
error: Invalid value for '--log-stderr-threshold <log-stderr-threshold>': error parsing level filter: expected one of "off", "error", "warn", "info", "debug", "trace", or a number 0-5
```

Verify that the Docker image works:
```
$ podman build -t test -f docker/Dockerfile.binary .
<snip>
[2/2] STEP 7/7: ENTRYPOINT ["/usr/bin/tini"]
[2/2] COMMIT test
--> b0d9e6f3ccf
Successfully tagged localhost/test:latest
b0d9e6f3ccfeba1b36e8d8f81e0985861efdf97879651f0f54fee0a6f3eecacb
$ docker run test solver
2021-10-11T11:18:12.844Z  INFO solver: running solver with validated Arguments {
<snip>
```
